playbooks/migration_precheck.yml

```diff
@@ -0,0 +1,183 @@
+---
+- name: Check Inventory file
+  hosts: localhost
+  gather_facts: false
+  tasks:
+    - name: Check the presence of Controller in the inventory file
+      fail:
+        msg:  "Please add Kraft Controller hosts in the inventory file for migration."
+      when:
+        - not (('kafka_controller' in groups.keys() and groups['kafka_controller'] | length > 0) | bool)
+
+    - name: Check the presence of Zookeeper in the inventory file
+      fail:
+        msg:  "Please add Zookeeper hosts in the inventory file for migration."
+      when:
+        - not (('zookeeper' in groups.keys() and groups['zookeeper'] | length > 0) | bool)
+
+    - name: Check the migration flag in the inventory file
+      fail:
+        msg:  "Please enable Kraft Migration flag in the inventory file for migration."
+      when:
+        - not (kraft_migration | default(false) | bool)
+
+- name: Check Current Confluent Version
+  hosts: zookeeper,kafka_broker
+  gather_facts: false
+  any_errors_fatal: true
+  tasks:
+    - name: Gather OS Facts
+      setup:
+        # Only gathers items in list, filters out the rest
+        filter: ansible_os_family
+        gather_subset:
+          - '!all'
+
+    - import_role:
+        name: variables
+
+    - name: Load override.conf
+      slurp:
+        src: "{{ zookeeper.systemd_override }}"
+      register: slurped_override
+      when: installation_method == "archive" and "zookeeper" in group_names
+
+    - name: Set current_confluent_version variable Zookeeper - Archive Installer
+      set_fact:
+        current_confluent_version: "{{ (slurped_override.content|b64decode) .split('\n') |
+              select('match', '^ExecStart=' + config_base_path + '/confluent-(.*)/bin/zookeeper-server-start ' + zookeeper.config_file) |
```

Comment:
Should we use path_join filter here to avoid issues which can be caused if someone defines the variable config_base_path as '/opt/confluent/' in the inventory file

Reply:
Let's start another thread for this since this is independent of migration and if we need to update this, we'll need to make the changes starting from 6.x

Reply:
makes sense